DigitalLearningSolutions.Data/Services/PasswordResetService.cs

```diff
@@ -112,6 +144,21 @@ TimeSpan expiryTime
             );
         }
 
+        public void SendWelcomeEmailsToDelegates(IEnumerable<DelegateUser> delegateUsers, DateTime emailDate, string baseUrl)
+        {
+            var emails = delegateUsers.Select(delegateUser => delegateUser.EmailAddress!).ToList();
+            
+            foreach (var email in emails)

```

Comment:
Just realised this now, but I think it would be better to do this in one query - maybe we can have a method GenerateAndScheduleDelegateWelcomeEmails?

Reply:
I can't find a way to do it (multiple/bulk insert) with one query using Dapper, unless we build the query string using the raw values. Best I could do is pass in the list of params objects (one params object for each email) to Execute and it automatically loops through them and runs the query for each one. These get executed within a single open connection so it's better than before. In fact because we are using IEnumerables, the inserting of the ResetPasswordIDs is also done at the same time.

Reply:
This is perfect :)